geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -94,6 +102,34 @@ public RedisSet(int size) {
     return diff;
   }
 
+  private static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      Set<byte[]> diff) {
+    if (regionProvider.getRedisData(destinationKey).getType() != REDIS_SET) {
+      regionProvider.getDataRegion().remove(destinationKey);
```

Comment:
In this "if" block you can set "destinationKeyExists" to true since you are doing a remove.
Then add an "else" to this block that does the "exists" call on line 111.
Also instead of calling both exists and getTypedRedisData, change the code to just call getTypedRedisData. If it returns null that is the same as exists returning false. We should only need to do one region op to find the entry; then we modify it, and then do a region op to store the changes.